web/tests/filter.test.js

```diff
@@ -1818,7 +1818,9 @@ test("try_adjusting_for_moved_with_target", ({override}) => {
     filter = new Filter(terms);
     filter.try_adjusting_for_moved_with_target();
     assert.deepEqual(filter.requires_adjustment_for_moved_with_target, false);
-    assert.deepEqual(filter.terms(), terms);
+    assert.deepEqual(filter.terms(), [
+        {operator: "dm", operand: "user3@zulip.com", negated: false},
+    ]);
```

Comment:
After correcting of narrow, `with` operator is removed by `_fix_redundant_is_private`